### ISSUE TITLE
[BUG] DataGrid does not update rows

### ISSUE THREAD

**Issue by unrealsolver** [formio/formio.js#1830]
## Environment
Please provide as many details as you can:

- Hosting type
  - [x] Form.io
  - [x] Local deployment
- Formio.js version: ^4.2.0-rc.5 and master
- Frontend framework: vanilla and react
- Browser: firefox, chrome
- Browser version: latest

## Steps to Reproduce
1. Create an empty form.
2. Add DataGrid
3. Add text input into the DataGrid
4. Add any other component (button, for example), into the form (not the DataGrid)
5. set submission which contains data for the DataGrid and consist of 2+ array items.

## Expected behavior
The form, which contains a DataGrid and the second component (button). The DataGrid contains several rows with inputs populated according to the provided data.
![expected](https://user-images.githubusercontent.com/1101771/66398206-e8368c80-e9cc-11e9-939b-ac9b17051b99.png)

## Observed behavior
Same as expected, but the DataGrid contains only one input with data. If I press 'Add One'. the rest lines reveals and also empty lines get added at the bottom.
![fio-observed](https://user-images.githubusercontent.com/1101771/66398227-ee2c6d80-e9cc-11e9-88cb-8d29a80ec088.png)

When clicked 'Add Another'
![fio-observed-when-clicked](https://user-images.githubusercontent.com/1101771/66398261-fd132000-e9cc-11e9-9773-cbd37e2765b9.png)

## Example

https://jsfiddle.net/rqg63wm7/

For code or form JSON, please enclose in a code block:

```js
Formio.createForm(document.getElementById('formio'), {
  components: [
    {
      label: 'Children',
      key: 'data',
      type: 'datagrid',
      input: true,
      components: [{
        label: 'First Name',
        key: 'text',
        type: 'textfield',
        input: true,
      }]
    }, {
      label: 'sub',
      key: 'sub',
      type: 'button',
      input: true,
    }
  ]
}).then(function(form) {
  // Provide a default submission.
  form.submission = {
    data: {
      data: [
        { text: 'Joe', },
        { text: 'Mary', }
      ]
    }
  }
})
```
# Important notes
If you remove the second component (button), the bug will not reproduce. It doesn't seem to depend on which type the second component has, it could be reproduced with input too, or with any other.

# Termporal fix
I added `update && this.redraw()` at the end of `DataGrid::setValue()`. I'm pretty sure it is not optimal, but it is the best that I can do in short term without knowing code base well.

**Comment by jhen1422** [comment]
@unrealsolver 
We are aware of this issue and have a ticket in our 4x backlog. Please note, we welcome pull requests in this library. If you need this item expedited, please feel free to contact support@form.io for pricing.

**Comment by unrealsolver** [comment]
Thanks! Anyway, I going to keep using formio and possible will be able to fix some bugs. Can I read a 'developer's/contributor's guide' somewhere?

**Comment by wag110894** [comment]
This issue has been resolved in the latest version 4.3.3.
